qiskit/providers/basic_provider/basic_provider.py

```diff
@@ -11,36 +11,40 @@
 # that they have been altered from the originals.
 
 
-"""Provider for Basic Aer simulator backends."""
+"""Provider for test simulator backends, formerly known as `BasicAer`."""
 
+from __future__ import annotations
+
+from collections.abc import Callable
 from collections import OrderedDict
+from typing import Type
+
 import logging
 
 from qiskit.exceptions import QiskitError
+from qiskit.providers.backend import Backend
 from qiskit.providers.provider import ProviderV1
 from qiskit.providers.exceptions import QiskitBackendNotFoundError
 from qiskit.providers.providerutils import resolve_backend_name, filter_backends
 
-from .qasm_simulator import QasmSimulatorPy
-from .statevector_simulator import StatevectorSimulatorPy
-from .unitary_simulator import UnitarySimulatorPy
+from .basic_simulator import BasicSimulator
 
 
 logger = logging.getLogger(__name__)
 
-SIMULATORS = [QasmSimulatorPy, StatevectorSimulatorPy, UnitarySimulatorPy]
+SIMULATORS = [BasicSimulator]
 
 
-class BasicAerProvider(ProviderV1):
-    """Provider for Basic Aer backends."""
+class BasicProvider(ProviderV1):
```

Comment:
I haven't found any `ProviderV2` class implementation. I was also confused by this, but the `IBMProvider` in `qiskit-ibm-provider` also inherits from `ProviderV1`. My understanding here is that the provider versions have nothing to do with the backend versions, and that instead of increasing provider versions what we have are different kinds of providers (like the runtime service).